esmvaltool/diag_scripts/land_carbon_cycle/diag_zonal_correlation.py

```diff
@@ -0,0 +1,452 @@
+             label=diag_config['obs_info']['source_label'])
+    sp1.fill_betweenx(lats_obs.points,
+                      r_tau_ctotal_tas_5.data,
+                      r_tau_ctotal_tas_95.data,
+                      facecolor='grey',
+                      alpha=0.40)
+
+    # tau-pr correlations
+    sp2 = plt.subplot(1, 2, 2)
+
+    # get the observations out of the dictionary
+    obs_var = diag_config.get('obs_variable')[1]
+    r_tau_ctotal_pr = zonal_correlation_obs[obs_var]
+    r_tau_ctotal_pr_5 = zonal_correlation_obs[obs_var + '_5']
+    r_tau_ctotal_pr_95 = zonal_correlation_obs[obs_var + '_95']
+    _fix_axis(obs_var, fig_config)
+
+    # plot the correlations from observation
+    sp2.plot(r_tau_ctotal_pr.data,
+             lats_obs.points,
+             color='k',
+             lw=1.1,
+             label=diag_config['obs_info']['source_label'])
+    sp2.fill_betweenx(lats_obs.points,
+                      r_tau_ctotal_pr_5.data,
+                      r_tau_ctotal_pr_95.data,
+                      facecolor='grey',
+                      alpha=0.40)
+
+    # PLOTTING for models
+
+    # loop over models and plot zonal correlations
+    for row_m in range(nmodels):
+        row_mod = models[row_m]
+        r_mod = zonal_correlation_mod[row_mod]['data']
+        lats_mod = zonal_correlation_mod[row_mod]['latitude']
+        r_tau_tas_c_pr_mod = r_mod[:, 0]
+        r_tau_pr_c_tas_mod = r_mod[:, 1]
+        if row_mod in ['MultiModelMedian', 'MultiModelMean']:
+            sp1.plot(np.ma.masked_equal(r_tau_tas_c_pr_mod, np.nan),
+                     lats_mod.points,
+                     lw=1.1,
+                     color='blue',
+                     label=row_mod)
+            sp2.plot(np.ma.masked_equal(r_tau_pr_c_tas_mod, np.nan),
+                     lats_mod.points,
+                     lw=1.1,
+                     color='blue',
+                     label=row_mod)
+        else:
+            sp1.plot(np.ma.masked_equal(r_tau_tas_c_pr_mod, np.nan),
+                     lats_mod.points,
+                     lw=0.3,
+                     label=row_mod)
+            sp2.plot(np.ma.masked_equal(r_tau_pr_c_tas_mod, np.nan),
+                     lats_mod.points,
+                     lw=0.3,
+                     label=row_mod)
+
+    # normalized mean correlations from model
+
+    # remove the multimodel estimates
+    models = list(zonal_correlation_mod.keys())
+    for _mm in multiModels:
+        if _mm in models:
+            models.remove(_mm)
+
+    nmodels = len(models)
+
+    r_tau_pr_c_tas_all = np.ones((len(lats_obs.points), nmodels)) * np.nan
+    r_tau_tas_c_pr_all = np.ones((len(lats_obs.points), nmodels)) * np.nan
+    for row_m in range(nmodels):
+        row_mod = models[row_m]
+        r_mod = zonal_correlation_mod[row_mod]['data']
+        lats_mod = zonal_correlation_mod[row_mod]['latitude']
+        r_tau_tas_c_pr_all[:, row_m] = r_mod[:, 0]
+        r_tau_pr_c_tas_all[:, row_m] = r_mod[:, 1]
```

Comment:
these variable names are monsters :japanese_ogre:

Reply:
Thank you ;-) But, they are as they are.